bundle/src/main/java/com/adobe/acs/commons/properties/filter/CapturingResponseWrapper.java

```diff
@@ -0,0 +1,114 @@
+/*
+ * #%L
+ * ACS AEM Commons Bundle
+ * %%
+ * Copyright (C) 2020 Adobe
+ * %%
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ * #L%
+ */
+
+package com.adobe.acs.commons.properties.filter;
+
+import javax.servlet.ServletOutputStream;
+import javax.servlet.WriteListener;
+import javax.servlet.http.HttpServletResponse;
+import javax.servlet.http.HttpServletResponseWrapper;
+import java.io.ByteArrayOutputStream;
+import java.io.IOException;
+import java.io.OutputStreamWriter;
+import java.io.PrintWriter;
+
+/**
+ * Response Wrapper used to allow for full output of the current response data so that it can be manipulated
+ * and rewritten.
+ */
+public class CapturingResponseWrapper extends HttpServletResponseWrapper {
+
+    private final ByteArrayOutputStream capture;
+    private ServletOutputStream output;
+    private PrintWriter writer;
+
+    public CapturingResponseWrapper(HttpServletResponse response) {
+        super(response);
+        capture = new ByteArrayOutputStream(response.getBufferSize());
+    }
+
+    @Override
+    public ServletOutputStream getOutputStream() {
+        if (writer != null) {
+            throw new IllegalStateException(
+                    "getWriter() has already been called on this response.");
+        }
+
+        if (output == null) {
+            output = new ServletOutputStream() {
+                @Override
+                public void write(int b) throws IOException {
+                    capture.write(b);
+                }
+
+                @Override
+                public void flush() throws IOException {
+                    capture.flush();
+                }
+
+                @Override
+                public void close() throws IOException {
+                    capture.close();
+                }
+
+                @Override
+                public boolean isReady() {
+                    return false;
```

Comment:
Through all my testing I haven't run into issues that seem to point this to being an issue, but I can't say for sure if _no one_ will. I'm not positive what this is supposed to be doing

Reply:
@adamcin do you have any idea?

Reply:
@id-keenan any reason we dont return `response.isReady()`?

Reply:
actually nvm - this is a `ServletOutputStream` instead of a `HttpServletResponse` so ignore my comment.  Carry on.

Reply:
Reading on `https://tomcat.apache.org/tomcat-8.0-doc/servletapi/javax/servlet/ServletOutputStream.html#isReady()` - "Checks if a non-blocking write will succeed. If this returns false, it will cause a callback to WriteListener.onWritePossible() when the buffer has emptied. If this method returns false no further data must be written until the contain calls WriteListener.onWritePossible()."


I certainly cant claim to understand how this all works, but it seems that since we're making `setWriteListener` a noop below then I think we should maybe set this function to `true`.

Obviously it seems to be working as-is right now, so maybe it's not a big deal.  However, "true" feels more correct.

Reply:
I'll update it to `true` locally and see if everything still seems to work